cmd/cli/handler_janitor_test.go

```diff
@@ -0,0 +1,247 @@
+package cli
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	"github.com/ory/hydra/internal/testhelpers"
+	"github.com/ory/x/cmdx"
+	"github.com/ory/x/configx"
+	"github.com/ory/x/flagx"
+
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	keepIfYounger          = "keep-if-younger"
+	accessLifespan         = "access-lifespan"
+	refreshLifespan        = "refresh-lifespan"
+	consentRequestLifespan = "consent-request-lifespan"
+	onlyTokens             = "tokens"
+	onlyRequests           = "requests"
+)
+
+func newJanitorCmd() *cobra.Command {
+	janitor := newJanitorHandler()
+	JanitorCmd := &cobra.Command{
+		Use:  "janitor",
+		RunE: janitor.Purge,
+		Args: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 0 &&
+				!flagx.MustGetBool(cmd, "read-from-env") &&
+				len(flagx.MustGetStringSlice(cmd, "config")) == 0 {
+
+				fmt.Printf("%s\n", cmd.UsageString())
+				return fmt.Errorf("%s\n%s\n%s\n",
+					"A DSN is required as a positional argument when not passing any of the following flags:",
+					"- Using the environment variable with flag -e, --read-from-env",
+					"- Using the config file with flag -c, --config")
+			}
+
+			if (!flagx.MustGetBool(cmd, onlyTokens) && !flagx.MustGetBool(cmd, onlyRequests)) || (flagx.MustGetBool(cmd, onlyTokens) && flagx.MustGetBool(cmd, onlyRequests)) {
```

Comment:
Does this mean we can't use `--tokens --requests` together? If so, why?

Reply:
I think I misunderstood your review here https://github.com/ory/hydra/pull/2381#discussion_r597475800

I can re-add if both are used together.

Reply:
Yeah, it should be possible to clean up everything with just one command!